core/src/test/java/org/apache/iceberg/TestRowLineageMetadata.java

```diff
@@ -241,85 +257,110 @@ public void testDeletes() {
+            .withReferencedDataFile(file.location())
+            .build();
+
+    table.newRowDelta().addDeletes(deletes).commit();
+
+    // Delete file records do not count toward added rows
+    assertThat(table.currentSnapshot().firstRowId()).isEqualTo(30);
+    assertThat(table.currentSnapshot().addedRows()).isEqualTo(0);
+    assertThat(table.ops().current().nextRowId()).isEqualTo(30);
   }
 
   @TestTemplate
-  public void testEnableRowLineageViaProperty() {
-    assumeThat(formatVersion).isGreaterThanOrEqualTo(TableMetadata.MIN_FORMAT_VERSION_ROW_LINEAGE);
+  public void testEqualityDeletes() {
+    assumeThat(formatVersion).isGreaterThanOrEqualTo(ROW_LINEAGE_MIN_FORMAT_VERSION);
 
     TestTables.TestTable table =
         TestTables.create(
             tableDir, "test", TEST_SCHEMA, PartitionSpec.unpartitioned(), formatVersion);
 
-    assertThat(table.ops().current().rowLineageEnabled()).isFalse();
+    assertThat(table.ops().current().nextRowId()).isEqualTo(0L);
 
-    // No-op
-    table.updateProperties().set(TableProperties.ROW_LINEAGE, "false").commit();
-    assertThat(table.ops().current().rowLineageEnabled()).isFalse();
+    DataFile file = fileWithRows(30);
 
-    // Enable row lineage
-    table.updateProperties().set(TableProperties.ROW_LINEAGE, "true").commit();
-    assertThat(table.ops().current().rowLineageEnabled()).isTrue();
+    table.newAppend().appendFile(file).commit();
 
-    // Disabling row lineage is not allowed
-    assertThatThrownBy(
-            () -> table.updateProperties().set(TableProperties.ROW_LINEAGE, "false").commit())
-        .isInstanceOf(IllegalArgumentException.class)
-        .hasMessageContaining("Cannot disable row lineage once it has been enabled");
+    assertThat(table.currentSnapshot().firstRowId()).isEqualTo(0);
+    assertThat(table.ops().current().nextRowId()).isEqualTo(30);
+
+    DeleteFile deletes =
+        FileMetadata.deleteFileBuilder(PartitionSpec.unpartitioned())
+            .ofEqualityDeletes(table.schema().findField("x").fieldId())
+            .withFormat(FileFormat.PARQUET)
+            .withFileSizeInBytes(100)
+            .withRecordCount(10)
+            .withPath("deletes.parquet")
+            .withReferencedDataFile(file.location())
+            .build();
+
+    table.newRowDelta().addDeletes(deletes).commit();
 
-    // No-op
-    table.updateProperties().set(TableProperties.ROW_LINEAGE, "true").commit();
-    assertThat(table.ops().current().rowLineageEnabled()).isTrue();
+    // Delete file records do not count toward added rows
+    assertThat(table.currentSnapshot().firstRowId()).isEqualTo(30);
+    assertThat(table.currentSnapshot().addedRows()).isEqualTo(0);
+    assertThat(table.ops().current().nextRowId()).isEqualTo(30);
   }
 
   @TestTemplate
-  public void testEnableRowLineageViaPropertyAtTableCreation() {
-    assumeThat(formatVersion).isGreaterThanOrEqualTo(TableMetadata.MIN_FORMAT_VERSION_ROW_LINEAGE);
+  public void testReplace() {
```

Comment:
Updated this test because `DataOperation.REPLACE` row counts are now handled correctly.

Reply:
Rolled this back. I'll fix it when I update how first-row-id gets assigned.